tests/vanilla/dependency.test.tsx

```diff
@@ -274,3 +274,43 @@ it('refreshes deps for each async read', async () => {
   resolve.splice(0).forEach((fn) => fn())
   expect(values).toEqual([0, 1])
 })
+
+it('handles errors in async atoms', async () => {
+  const errorAtom = atom(async () => {
+    throw new Error('Test error')
+  })
+
+  const store = createStore()
+
+  await expect(store.get(errorAtom)).rejects.toThrow('Test error')
+})
+
+it('handles complex dependency chains', async () => {
+  const baseAtom = atom(1)
+  const derived1 = atom((get) => get(baseAtom) * 2)
+  const derived2 = atom((get) => get(derived1) + 1)
+  const asyncDerived = atom(async (get) => {
+    const value = get(derived2)
+    await new Promise((resolve) => setTimeout(resolve, 10))
```

Comment:
We don't use timers in the test.
Please do something like this:
https://github.com/pmndrs/jotai/blob/9bd8ed7b0e523af5366f6b762104241a0bb71203/tests/vanilla/dependency.test.tsx#L11

Reply:
@dai-shi Appreciated for indication and this below is adjustment which follows comment here.
[6186a57](https://github.com/pmndrs/jotai/pull/2759/commits/6186a5731564688e3cd2c2e317e67b5902d9ac0c)